crates/evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -758,20 +758,21 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
         if let Some(prank) = &self.prank {
             if data.journaled_state.depth() == prank.depth {
                 data.env.tx.caller = h160_to_b160(prank.prank_origin);
-            }
-            if prank.single_call {
-                std::mem::take(&mut self.prank);
+
+                if prank.single_call {
+                    std::mem::take(&mut self.prank);
+                }
```

Comment:
Let's add docs on the rationale as to why this is applied iff the outer `if` runs—it's a change on the previous behavior and I'm a bit apprehensive it might "fix" (or break) behavior that has come to be expected. Will test this locally